tests/system_tests_edge_router.py

```diff
@@ -1895,6 +1895,10 @@ def test_51_link_route_proxy_configured(self):
         if self.skip['test_51'] :
             self.skipTest("Test skipped during development.")
 
+        # The previous test waited for address CfgLinkRoute1 on router INT_B
+        # We will wait for address CfgLinkRoute1 to be unsubscribed
+        self.INT_B.wait_address_unsubscribed("CfgLinkRoute1")
```

Comment:
Depending on the specific order in which the tests are run sounds dangerous. I think that this line should be part of the previous test.

It is probably sufficient to put it at the end. That way it would be skipped if the test failed before it got to it, but that should be rare,...

There is currently not a good way to have a per-test tear-down in system-tests. One could be developed... Maybe a decorator on the test method? Or go all-in with pytest and use fixtures, which do have their teardowns? The least disruptive thing would be to have a list in the test class, `self.addresses_requiring_unsubscribe`, which would be cleaned in the shared tear_down code. Tests would populate it with addresses that require this treatment.

Reply:
Another way, maybe not quick, but certainly dirty, would be to require that each test suffixes all addresses it uses with it's own name. The name can be found by introspecting the Python stack, or maybe the unittest module already knows the name of the currently running test. Randomly generated string would work as well.

The problem is that these unique addresses cannot be preconfigured ar the beginning, each test has to create them through management.